src/day-1-toy/12_args.py

```diff
@@ -20,28 +32,34 @@
 
 a = [7, 6, 5, 4]
 
-# What thing do you have to add to make this work?
-print(f2(a))    # Should print 22
+# What thing do you have to add to make this work? use the * in front 
+print(f2(*a))    # Should print 22
 
 # Write a function f3 that accepts either one or two arguments. If one argument,
 # it returns that value plus 1. If two arguments, it returns the sum of the
 # arguments. Google "python default arguments" for a hint.
 
-#def f3(...
+def f3(a, b=1):
+    return a + b
 
 print(f3(1, 2))  # Should print 3
 print(f3(8))     # Should print 9
 
 
 # Write a function f4 that accepts an arbitrary number of keyword arguments and
-# prints ouf the keys and values like so:
+# prints out the keys and values like so:
 #
 # key: foo, value: bar
 # key: baz, value: 12
 #
 # Google "python keyword arguments".
+# https://www.saltycrane.com/blog/2008/01/how-to-use-args-and-kwargs-in-python/
+# https://www.digitalocean.com/community/tutorials/how-to-use-args-and-kwargs-in-python-3
+# kwargs means key word arguments
 
-#def f4(...
+def f4(**kwargs):
+    for key, value in kwargs.items():
+        print(f"key: {key}, value: {kwargs[key]}")
```

Comment:
Since you've destructured both the `key` and `value` from `kwargs`, you may reference the value through `value`:
```
print(f"key: {key}, value: {value}")
```